executor.py

```diff
@@ -1,7 +1,114 @@
-from jina import Executor, DocumentArray, requests
+from jina import Executor, requests
+from typing import Optional, Dict, Any, List
+from docarray import DocumentArray
+from jina.logging.logger import JinaLogger
 
 
 class ElasticSearchIndexer(Executor):
-    @requests
-    def foo(self, docs: DocumentArray, **kwargs):
-        pass
+    def __init__(
+        self,
+        hosts: Optional[str] = 'http://localhost:9200',
```

Comment:
it is weird to have `hosts` but have a single string

Reply:
And is it actually `Optional`? What is the behavior when None is given?

Reply:
So the ElasticSearch client exposes a parameter `hosts` with the following type: `Union[str, List[Union[str, Mapping[str, Union[str, int]]]], None]`. This means it can be either a single node (str) or a list of nodes with custom configurations.
I'm adding a slight modification in docarray to reflect this type hint:
https://github.com/jina-ai/docarray/pull/333

Reply:
no more optional